apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/CoreConfiguration.java

```diff
@@ -516,6 +519,10 @@ public boolean isInstrument() {
         return instrument.get();
     }
 
+    public List<ConfigurationOption<?>> getInstrumentationOptions() {
+        return Arrays.asList(instrument, traceMethods, disabledInstrumentations);
```

Comment:
`traceMethods` and `disabledInstrumentations` are tested, but seems that dynamically changing `instrument` is not tested.